pkg/datapath/loader/base.go

```diff
@@ -107,6 +110,42 @@ func writePreFilterHeader(preFilter *prefilter.PreFilter, dir string) error {
 	return fw.Flush()
 }
 
+// Return the list of interfaces with egress masquerading.
+// - If option.Config.Masquerade is disabled, return an empty slice.
+// - If option.Config.EgressMasqueradeInterfaces is set to a simple interface
+//   name, return its value.
+// - If it is empty, return the list of all interfaces.
+// - If it has a '+' suffix, return the names of all matching interfaces.
+func getEgressMasqueradeInterfaces() ([]string, error) {
+	var ifaces []string
+	if !option.Config.Masquerade {
+		return ifaces, nil
+	}
+
+	egMasqIfOpt := &option.Config.EgressMasqueradeInterfaces
+	if *egMasqIfOpt != "" && !strings.HasSuffix(*egMasqIfOpt, "+") {
+		ifaces = append(ifaces, option.Config.EgressMasqueradeInterfaces)
+		return ifaces, nil
+	}
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		return ifaces, err
+	}
+	if *egMasqIfOpt == "" {
+		for _, l := range links {
+			ifaces = append(ifaces, l.Attrs().Name)
+		}
+	} else {
+		for _, l := range links {
+			if strings.HasPrefix(l.Attrs().Name, (*egMasqIfOpt)[:len(*egMasqIfOpt)-1]) {
+				ifaces = append(ifaces, l.Attrs().Name)
+			}
+		}
```

Comment:
With the above change of the declaration of `egMasqIfOpt`:

```suggestion
		egMasqIfOpt := strings.TrimSuffix(egMasqIfOpt, "+")
		for _, l := range links {
			if strings.HasPrefix(l.Attrs().Name, egMasqIfOpt) {
				ifaces = append(ifaces, l.Attrs().Name)
			}
		}
```

IMO this makes it easier to follow that the `+` is stripped.